config/schema/synonyms.yml

```diff
@@ -31,11 +31,19 @@
 - index: hazardous substances, dangerous substances
 - index: holiday pay, holiday entitlement
 - index: pension forecast, pension statement
-- index: practical driving test, practical test
-- index: provisional driving licence, provisional licence
 - index: tariff classification, trade tariff
 - index: tariff codes, commodity codes
 
+# Treat synonyms of differing lengths the same
+
+# Multiword synonyms "play havoc" with term positions, so avoid creating index rules that change the phrase length.
+# See: https://www.elastic.co/guide/en/elasticsearch/guide/current/multi-word-synonyms.html
+# If a phrase expands to a longer phrase, extra words will be highlighted in search descriptions.
+- both: practical test, practical tests => practical test, practical_driving_test
```

Comment:
I take it we don't need to use the special character in front of the underscored words to avoid it being stemmed?

Reply:
That's a good point - I don't expect this to be stemmed atm, but I'll add it in to make it clear what's going on.

Reply:
On second thoughts I'm going to skip this and just put in a comment, because we actually took out the protwords filter that does this: https://github.com/alphagov/rummager/commit/4f0013f43fff0e85a5e904d07fc53697fe7d0a3a

These terms do go through the stemmer but it shouldn't cause any issues since it's not stemming to an actual word stem.

```
elyzer --es 'http://dev.gov.uk:9200' --index 'detailed-2018-01-05t13:50:50z-6d703df4-b122-4a6a-ad6f-a55ebf635853' --text 'practical test' --analyzer 'with_index_synonyms'
CHAR_FILTER: normalize_quotes
{0:practical}	{1:test}
CHAR_FILTER: strip_quotes
{0:practical}	{1:test}
TOKENIZER: standard
{0:practical}	{1:test}
TOKEN_FILTER: standard
{0:practical}	{1:test}
TOKEN_FILTER: asciifolding
{0:practical}	{1:test}
TOKEN_FILTER: lowercase
{0:practical}	{1:test}
TOKEN_FILTER: index_synonym
{0:practical,practical_driving_test}	{1:test}
TOKEN_FILTER: stop
{0:practical,practical_driving_test}	{1:test}
TOKEN_FILTER: stemmer_override
{0:practical,practical_driving_test}	{1:test}
TOKEN_FILTER: stemmer_english
{0:practical,practical_driving_test}	{1:test}
```

```
elyzer --es 'http://dev.gov.uk:9200' --index 'detailed-2018-01-05t13:50:50z-6d703df4-b122-4a6a-ad6f-a55ebf635853' --text 'provisional licence' --analyzer 'with_index_synonyms'
CHAR_FILTER: normalize_quotes
{0:provision}	{1:licenc}
CHAR_FILTER: strip_quotes
{0:provision}	{1:licenc}
TOKENIZER: standard
{0:provisional}	{1:licence}
TOKEN_FILTER: standard
{0:provisional}	{1:licence}
TOKEN_FILTER: asciifolding
{0:provisional}	{1:licence}
TOKEN_FILTER: lowercase
{0:provisional}	{1:licence}
TOKEN_FILTER: index_synonym
{0:provisional,provisional_driving_licence}	{1:licence}
TOKEN_FILTER: stop
{0:provisional,provisional_driving_licence}	{1:licence}
TOKEN_FILTER: stemmer_override
{0:provisional,provisional_driving_licence}	{1:licence}
TOKEN_FILTER: stemmer_english
{0:provision,provisional_driving_lic}	{1:licenc}
```